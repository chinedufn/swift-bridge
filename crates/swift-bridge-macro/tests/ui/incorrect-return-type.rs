//! # To Run
//! cargo test -p swift-bridge-macro -- ui trybuild=incorrect-return-type.rs

pub struct SomeType;

#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        type SomeType;

        #[swift_bridge(rust_name = "some_function")]
        fn fn1() -> SomeType;
        #[swift_bridge(rust_name = "another_function")]
        fn fn2() -> SomeType;
    }
}

fn some_function() -> &'static SomeType {
    &SomeType
}

fn another_function() -> Option<SomeType> {
    None
}

fn main() {}
