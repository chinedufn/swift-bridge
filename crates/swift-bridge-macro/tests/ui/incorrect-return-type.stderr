error[E0308]: mismatched types
 --> tests/ui/incorrect-return-type.rs:6:1
  |
6 | #[swift_bridge::bridge]
  | ^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected struct `SomeType`, found `&SomeType`
  | expected due to this
  |
  = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/ui/incorrect-return-type.rs:6:1
  |
6 | #[swift_bridge::bridge]
  | ^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected struct `SomeType`, found enum `Option`
  | expected due to this
  |
  = note: expected struct `SomeType`
               found enum `Option<SomeType>`
  = note: this error originates in the attribute macro `swift_bridge::bridge` (in Nightly builds, run with -Z macro-backtrace for more info)
